File: flutter/lib/src/cheyalpaadugal/niril_podhu/tharavuru/seluthi_vagai.dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

/// Type-safe payment mode enum with localized display names.
/// Stored as lowercase string in DB (e.g. 'cash', 'upi', 'bank_transfer').
enum SeluthiVagai {
  vangiMaatram, // Bank Transfer — வங்கி மாற்றம்
  upi,          // UPI
  panam,        // Cash — பணம்
  kaasoalai,    // Cheque — காசோலை
  attai,        // Card — அட்டை
}

extension SeluthiVagaiX on SeluthiVagai {
  /// The stored value in the database column.
  String get storedValue {
    switch (this) {
      case SeluthiVagai.panam:
        return 'cash';
      case SeluthiVagai.upi:
        return 'upi';
      case SeluthiVagai.vangiMaatram:
        return 'bank_transfer';
      case SeluthiVagai.kaasoalai:
        return 'cheque';
      case SeluthiVagai.attai:
        return 'card';
    }
  }

  /// Parse from stored string. Returns null for unknown values.
  static SeluthiVagai? fromStored(String? value) {
    if (value == null || value.isEmpty) return null;
    switch (value) {
      case 'cash':
        return SeluthiVagai.panam;
      case 'upi':
        return SeluthiVagai.upi;
      case 'bank_transfer':
        return SeluthiVagai.vangiMaatram;
      case 'cheque':
        return SeluthiVagai.kaasoalai;
      case 'card':
        return SeluthiVagai.attai;
      default:
        return null;
    }
  }

  /// Tamil display name.
  String get tamilLabel {
    switch (this) {
      case SeluthiVagai.panam:
        return 'பணம்';
      case SeluthiVagai.upi:
        return 'UPI';
      case SeluthiVagai.vangiMaatram:
        return 'வங்கி மாற்றம்';
      case SeluthiVagai.kaasoalai:
        return 'காசோலை';
      case SeluthiVagai.attai:
        return 'அட்டை';
    }
  }

  /// English display name.
  String get englishLabel {
    switch (this) {
      case SeluthiVagai.panam:
        return 'Cash';
      case SeluthiVagai.upi:
        return 'UPI';
      case SeluthiVagai.vangiMaatram:
        return 'Bank Transfer';
      case SeluthiVagai.kaasoalai:
        return 'Cheque';
      case SeluthiVagai.attai:
        return 'Card';
    }
  }

  /// Whether this payment mode requires a reference/transaction ID.
  bool get needsReference => this != SeluthiVagai.panam;

  /// Cupertino icon for each mode.
  IconData get icon {
    switch (this) {
      case SeluthiVagai.panam:
        return CupertinoIcons.money_dollar_circle;
      case SeluthiVagai.upi:
        return CupertinoIcons.device_phone_portrait;
      case SeluthiVagai.vangiMaatram:
        return CupertinoIcons.building_2_fill;
      case SeluthiVagai.kaasoalai:
        return CupertinoIcons.doc_text;
      case SeluthiVagai.attai:
        return CupertinoIcons.creditcard;
    }
  }

  /// Color for pill badge.
  Color badgeColor(bool isDark) {
    switch (this) {
      case SeluthiVagai.panam:
        return isDark ? Colors.green.shade300 : Colors.green.shade700;
      case SeluthiVagai.upi:
        return isDark ? Colors.purple.shade200 : Colors.purple.shade600;
      case SeluthiVagai.vangiMaatram:
        return isDark ? Colors.blue.shade200 : Colors.blue.shade700;
      case SeluthiVagai.kaasoalai:
        return isDark ? Colors.orange.shade200 : Colors.orange.shade700;
      case SeluthiVagai.attai:
        return isDark ? Colors.teal.shade200 : Colors.teal.shade700;
    }
  }
}

File: migrate/composeApp/src/commonMain/kotlin/com/elvan/noolachu/data/model/SeluthiVagai.kt
package com.elvan.noolachu.data.model

import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import com.elvan.noolachu.localization.K
import com.elvan.noolachu.localization.LanguageManager
import com.elvan.noolachu.localization.tr
import com.elvan.noolachu.localization.trWithLang
import com.elvan.noolachu.ui.navigation.MaterialSymbols

/**
 * Type-safe payment mode enum with localized display names.
 * Stored as lowercase string in DB (e.g. 'cash', 'upi', 'bank_transfer', 'cheque', 'card').
 */
enum class SeluthiVagai(
    val storedValue: String,
    val needsReference: Boolean
) {
    VANGI_MAATRAM("bank_transfer", true), // Bank Transfer — வங்கி மாற்றம்
    UPI("upi", true),                    // UPI
    PANAM("cash", false),                // Cash — காசு / பணம்
    KAASOALAI("cheque", true),           // Cheque — காசோலை
    ATTAI("card", true);                 // Card — அட்டை

    val icon: ImageVector
        get() = when (this) {
            PANAM -> MaterialSymbols.Rounded.Payments
            UPI -> MaterialSymbols.Rounded.QrCode
            VANGI_MAATRAM -> MaterialSymbols.Rounded.AccountBalance
            KAASOALAI -> MaterialSymbols.Rounded.ReceiptLong
            ATTAI -> MaterialSymbols.Rounded.CreditCard
        }

    fun labelString(langCode: String = LanguageManager.activeLanguageCode): String {
        return when (this) {
            PANAM -> K.kaasu.trWithLang(langCode)
            UPI -> "UPI"
            VANGI_MAATRAM -> K.vangiParimaatram.trWithLang(langCode)
            KAASOALAI -> K.kaasoalai.trWithLang(langCode)
            ATTAI -> K.attai.trWithLang(langCode)
        }
    }

    @Composable
    fun label(): String = labelString()

    fun badgeColor(isDark: Boolean): Color {
        return when (this) {
            PANAM -> if (isDark) Color(0xFF81C784) else Color(0xFF388E3C)
            UPI -> if (isDark) Color(0xFFCE93D8) else Color(0xFF7B1FA2)
            VANGI_MAATRAM -> if (isDark) Color(0xFF90CAF9) else Color(0xFF1976D2)
            KAASOALAI -> if (isDark) Color(0xFFFFCC80) else Color(0xFFF57C00)
            ATTAI -> if (isDark) Color(0xFF80CBC4) else Color(0xFF00796B)
        }
    }

    companion object {
        fun fromStored(value: String?): SeluthiVagai {
            if (value.isNullOrBlank()) return PANAM
            return when (value.trim().lowercase()) {
                "bank_transfer", "vangimaatram", "vangi_maatram" -> VANGI_MAATRAM
                "upi" -> UPI
                "cash", "panam", "roakkam" -> PANAM
                "cheque", "kaasoalai" -> KAASOALAI
                "card", "attai" -> ATTAI
                else -> PANAM
            }
        }

        fun allOptions(langCode: String = LanguageManager.activeLanguageCode): List<Pair<String, String>> = listOf(
            PANAM.name to PANAM.labelString(langCode),
            UPI.name to UPI.labelString(langCode),
            VANGI_MAATRAM.name to VANGI_MAATRAM.labelString(langCode),
            KAASOALAI.name to KAASOALAI.labelString(langCode),
            ATTAI.name to ATTAI.labelString(langCode)
        )
    }
}
